### PR TITLE
Make intermediate type for cumsum ScalarType::Int

### PR DESCRIPTION
**Pull request by abhudev**
Fixes #90635

### CODE REVIEW

**Review by DenisVieriu97 — APPROVED** [review]
Looks good

**Review by DenisVieriu97 — CHANGES_REQUESTED** [review]
@abhudev - can you error out in case of int64? The casting to int32 won't work

**Comment by abhudev** [comment]
@DenisVieriu97 int64 is now disallowed

File: aten/src/ATen/native/mps/operations/UnaryOps.mm [hunk 1]
```diff
@@ -262,11 +262,12 @@ void unary_op(const Tensor& self, const Tensor& output, std::string op_name, Una
     return;
   }
   auto input = dtype.has_value() ? self.to(dtype.value()) : self;
+  TORCH_CHECK(input.scalar_type() != ScalarType::Long, "Input type shouldn't be int64");
```
Comment (DenisVieriu97) [inline aten/src/ATen/native/mps/operations/UnaryOps.mm:265]: Can you please rephrase it to `MPS does not support cumsum op with int64 input` or something similar? We have a search pattern in `test_mps.py` to skip these tests with int64 issues

**Review by DenisVieriu97 — APPROVED** [review]
Looks good - added a minor comment, please take a look

File: aten/src/ATen/native/mps/operations/UnaryOps.mm (hunk 1 above)
Comment (abhudev) [reply aten/src/ATen/native/mps/operations/UnaryOps.mm]: Changed the error message, also added a test case
